src/cli.rs

```diff
@@ -0,0 +1,135 @@
+mod info;
+mod init;
+
+use std::env;
+use std::io;
+use std::path::PathBuf;
+use std::process;
+
+use clap::arg_enum;
+use structopt::StructOpt;
+
+use crate::errors::*;
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "TMSU",
+    about = "A tool for tagging your files and accessing them through a virtual filesystem"
+)]
+struct TmsuOptions {
+    // Externalize global options to a separate struct for convenience
+    #[structopt(flatten)]
+    global_opts: GlobalOptions,
+
+    #[structopt(subcommand)]
+    cmd: SubCommands,
+}
+
+#[derive(Debug, StructOpt)]
+pub struct GlobalOptions {
+    /// Use the specified database
+    #[structopt(short = "-D", long, env = "TMSU_DB", parse(from_os_str))]
+    database: Option<PathBuf>,
+
+    /// Colorize the output (auto/always/never)
+    #[structopt(long, default_value = "auto")]
+    color: ColorMode,
+}
+
+arg_enum! {
+    #[derive(Debug)]
+    enum ColorMode {
+        Auto,
+        Always,
+        Never,
+    }
+}
+
+#[derive(Debug, StructOpt)]
+enum SubCommands {
```

Comment:
To add a new subcommand we need to do the following:
* Add a `cli/subcommand.rs` file
* Add a `api/subcommand.rs` file
* List the new modules in `api.rs` and `cli.rs`
* Add a new entry in this enum
* Add a new entry in the `match` below (it won't compile if we forget this anyway)